File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLContactLinkNone.java
package com.github.badoualy.telegram.tl.api;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLContactLinkNone extends TLAbsContactLink {
    public static final int CONSTRUCTOR_ID = 0xfeedd3ad;

    public TLContactLinkNone() {
    }

    @Override
    public String toString() {
        return "contactLinkNone#feedd3ad";
    }

    @Override
    public int getConstructorId() {
        return CONSTRUCTOR_ID;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLContactLinkNone.kt
package com.github.badoualy.telegram.tl.api

/**
 * contactLinkNone#feedd3ad
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLContactLinkNone : TLAbsContactLink() {
    private val _constructor: String = "contactLinkNone#feedd3ad"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLContactLinkNone) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0xfeedd3ad.toInt()
    }
}
